File: binarySearch/searchBigSortedArray.java
public int searchBigSortedArray(ArrayReaer reader, int target){
	if(reader == null)
		return -1;
	int i = 1;
	while(reader.get(i-1) != null && reader.get(i-1) < target)
		i = i * 2;
	int left = 0;
	int right = i-1;
	while(left + 1 < right){
		int mid = left + (right - left)/2;
		if(reader.get(mid) == target)
			right = mid;
		if(reader.get(mid) < target)
			left = mid;
		else
			right = mid;
	}
	if(reader.get(left) == target)
		return left;
	if(reader.get(right) == target)
		return right;
	return -1;
}

File: binarySearch/searchBigSortedArray.py
def searchBigSortedArray(reader, target):
	right = 1
	while read.get[right-1] is not None and read.get(i-1) < target:
		right = right * 2
	right = right -1
	left = right -2
	while left + 1 < right:
		mid = left + (right - left) //2
		if reader.get(mid) >= target:
			right = mid
		else:
			left = mid

	if reader.get(left) == target:
		return left
	elif reader.get(right) == target:
		return right
	else:
		return -1
